gh-83714: Use "stx_" prefix for all os.statx_result members

vstinner opened the pull request:
<!--
Thanks for your contribution!
Please read this comment in its entirety. It's quite important.

# Pull Request title

It should be in the following format:

```
gh-NNNNNN: Summary of the changes made
```

Where: gh-NNNNNN refers to the GitHub issue number.

Most PRs will require an issue number. Trivial changes, like fixing a typo, do not need an issue.

# Backport Pull Request title

If this is a backport PR (PR made against branches other than `main`),
please ensure that the PR title is in the following format:

```
[X.Y] <title from the original PR> (GH-NNNNNN)
```

Where: [X.Y] is the branch name, for example: [3.13].

GH-NNNNNN refers to the PR number from `main`.

-->


<!-- gh-issue-number: gh-83714 -->
* Issue: gh-83714
<!-- /gh-issue-number -->


<!-- readthedocs-preview cpython-previews start -->
----
📚 Documentation preview 📚: https://cpython-previews--140432.org.readthedocs.build/

<!-- readthedocs-preview cpython-previews end -->

vstinner committed:
gh-83714: Use "stx_" prefix for all os.statx_result members

cmaloney commented on Doc/library/os.rst:
```diff
@@ -3412,7 +3412,7 @@ features:
    :class:`!statx_result` has all the attributes that :class:`~stat_result` has
-   on Linux, making it :term:`duck-typing` compatible, but
+   on Linux, but using ``stx_`` prefix instead of ``st_``.
    :class:`!statx_result` is not a subclass of :class:`~stat_result` and cannot
```
given the members no longer overlap can probably drop the "not a tuple" doc as well

vstinner replied:
I prefer to keep "cannot be used as a tuple".

cmaloney commented on Doc/library/os.rst:
I think this should list all the attributes now; doc having people translate in their head rather than see "here's the name I used" seems like it will break "search in page" finding the attributes.

vstinner replied:
Ok, I completed my PR to document all statx_members. It also clarifies which `stat_result` members are not part of `statx_result`:

* st_creator
* st_file_attributes
* st_flags
* st_fstype
* st_gen
* st_reparse_tag
* st_rsize
* st_type

cmaloney reviewed (commented):
I'm 👍 on this as its evolved. There's too many weird edged cases and just separating entirely makes simpler to reason around. 

I think if go forward with this all the `stx_` attributes should be explicitly documented rather than refer to `stat_result` ones of different name. I'm often in a codebase wondering "what is this" and search in page to find; the text note "all the stat members but with a `stx_` prefix" would break that

vstinner committed:
Document all statx_result members

Rename stx_birthtime to stx_btime, and rename stx_birthtime_ns to
stx_btime_ns.

vstinner committed:
Simplify posixmodule.c macros

Remove now redundant MEMBER parameter.

cmaloney commented on Doc/library/os.rst:
```diff
@@ -3412,132 +3412,252 @@ features:
+      :attr:`~statx_result.stx_mask`.
 
-      Direct I/O memory buffer alignment requirement.
+   .. attribute:: stx_ctime
```
(pondering / not sure if actually a good idea): Given we don't need compatibility why not just always do the `nanosecond` variant? That always has full resolution in the Python wrapper and can be rounded to seconds by the caller if wanted/needed.

vstinner replied:
It's less convenient to manipulate nanoseconds (int) than seconds (float). We may keep seconds by default and provide nanoseconds as an alternative.

cmaloney replied:
What about always returning https://docs.python.org/3/library/decimal.html? Then you get precision always with a count of seconds as just `int(st.stx_ctime)`.

vstinner replied:
I wrote [PEP 410 – Use decimal.Decimal type for timestamps](https://peps.python.org/pep-0410/) but it got rejected: https://mail.python.org/pipermail/python-dev/2012-February/116837.html

Later I wrote [PEP 564 – Add new time functions with nanosecond resolution](https://peps.python.org/pep-0564/) which was accepted in Python 3.7.

cmaloney replied:
Thanks for the links! Good pieces for me to know. I think that pretty comprehensively closes out this particular path of exploration :).

cmaloney reviewed (commented):
👍 overall; curious @jbosboom thoughts. Appreciate the fixes/tweaks to Linux kernel version availability.

vstinner committed:
Remove comparison to stat_result

vstinner committed:
Credit myself in os.statx() doc
